### PR TITLE
#114 custom changes_method is not getting called on object creation

### PR DESCRIPTION
**Pull request by sivagollapalli**

### CODE REVIEW

**Comment by dblock** [comment]
Lets start by fixing the build, you need to `rubocop -a` and possibly `rubocop --auto-gen-config`.

**Comment by dblock** [comment]
Please also update CHANGELOG. Rebase the branch. Thx!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 0.5.1 (Next)
 ------------
 
-* Your contribution here.
```
Comment (dblock) [inline CHANGELOG.md]: Put this back please.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+* [#144](https://github.com/aq1018/mongoid-history/pull/158): Can modify history tracker insertion on object creation
```
Comment (dblock) [inline CHANGELOG.md]: Make this look like other lines, it's missing your name.

**Comment by dblock** [comment]
Would be nice to squash your commits too.

**Comment by sivagollapalli** [comment]
@dblock Looks build has been passed for `mongoid 5`  but it has been failed for `v4` and `v3`. Do we need to fix that or shall leave it?

**Comment by dblock** [comment]
You need to fix it.

**Comment by sivagollapalli** [comment]
@dblock Made changes in PR and build fixed. Could you please check and let me know any changes needs to be made?

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 ------------
 
 * Your contribution here.
+* [#144](https://github.com/aq1018/mongoid-history/pull/158): Can modify history tracker insertion on object creation - [@sivagollapalli](https://github.com/sivagollapalli)
```
Comment (dblock) [inline CHANGELOG.md]: Missing period at the end.

File: spec/unit/trackable_spec.rb [hunk 4]
```diff
@@ -298,6 +298,22 @@ def my_changes
         expect(m).to receive(:my_changes).once.and_call_original
         m.save
       end
+
+      it 'should allow an alternate method to be specified on object creation' do
+        class CustomTracker < MyModel
```
Comment (dblock) [inline spec/unit/trackable_spec.rb]: note that this class isn't actually scoped to the `it`, if you really want to make it better, declare it via

```
let(:custom_tracker) { Class.new(MyModel) do ... }
```

I know we don't do it elsewhere, so I am not nitpicking on it.

**Comment by dblock** [comment]
Okay, one last minor thing ^^^, sorry to be nitpicky.

**Comment by dblock** [comment]
Merged, thank you.

**Comment by dblock** [comment]
This closes #114.

**Pull request merged**